nxcomp/src/Timestamp.cpp

```diff
@@ -44,34 +44,55 @@
 
 T_timestamp timestamp;
 
-//
-// The following functions all use the ctime
-// static buffer from the C library.
-//
-
-char *strTimestamp(const T_timestamp &ts)
+std::string strTimestamp(const T_timestamp &ts)
 {
-  char *ctime_now = ctime((time_t *) &ts.tv_sec);
+  std::string ret;
+
+  char ctime_now[26] = { };
+  bool err = true;
+
+#if HAVE_CTIME_S
+  errno_t retval = ::ctime_s(ctime_now, 26, static_cast<const time_t*>(&ts.tv_sec));
+
+  err = (retval != 0);
+#else
+  char *retval = ::ctime_r(static_cast<const time_t*>(&ts.tv_sec), ctime_now);
 
+  err = (!(retval));
+#endif
+
+  if (err)
+  {
+    std::cerr << "WARNING: converting time to string failed." << std::endl;
+  }
+
+  /* Replace newline at position 25 with a NULL byte. */
   ctime_now[24] = '\0';
 
-  return ctime_now;
+  ret = ctime_now;
```

Comment:
return ctime_now immediately, there's not need for ret AFAICS

Reply:
Actually, there was a good reason why I intended to use `ret`, which I have done wrong the first time around. `ret` should only be set to `ctime_now` if the call to `ctime_{r,s}` actually succeeded, else left as-is.